packages/url_launcher/url_launcher_android/android/src/main/java/io/flutter/plugins/urllauncher/UrlLauncher.java

```diff
@@ -118,6 +128,38 @@ public void closeWebView() {
     applicationContext.sendBroadcast(new Intent(WebViewActivity.ACTION_CLOSE));
   }
 
+  private static @NonNull Boolean openCustomTab(
+      @NonNull Context context, @NonNull Uri uri, @NonNull Bundle headersBundle) {
+    CustomTabsIntent customTabsIntent = new CustomTabsIntent.Builder().build();
+    customTabsIntent.intent.putExtra(Browser.EXTRA_HEADERS, headersBundle);
+    try {
+      customTabsIntent.launchUrl(context, uri);
+    } catch (ActivityNotFoundException ex) {
+      return false;
+    }
+    return true;
+  }
+
+  // Checks if headers contains a CORS restricted header.
+  //  https://developer.mozilla.org/en-US/docs/Glossary/CORS-safelisted_request_header
+  private static @NonNull Boolean containsRestrictedHeader(Map<String, String> headersMap) {
+    for (String key : headersMap.keySet()) {
+      switch (key.toLowerCase(Locale.US)) {
+        case "accept":
+          continue;
```

Comment:
Java supports multi-case blocks (just like Dart), so you don't need a separate statement for each supported header.

Reply:
Can't do multiple cases for single block because of source version being Java 8.
At best can do the scary, not so scary:
```
  case "accept":
        case "accept-language":
        case "content-language":
        case "content-type":
          continue;
```

Reply:
> At best can do the scary, not so scary:
> 
> ```
> case "accept":
> case "accept-language":
> case "content-language":
> case "content-type":
>   continue;
> ```

This is the construction I was referring to. I'm not sure why this would be scary; it's an extremely common pattern for switch statements in many languages.

Reply:
I prefer to avoid silent fallthroughs, but yeah pretty common

Reply:
Google's style guides ([including Java](https://google.github.io/styleguide/javaguide.html#s4.8.4-switch)) distinguish between stacked case labels (which are common, and don't have a history of being problematic) and what people generally think of as fallthrough, where there is both code and a fallthrough (and does have a history of being problematic, so is generally avoided).